### PR TITLE
Lots of things

### PR DESCRIPTION
**Pull request by JimmyPruitt**
This PR will be in favour of the two I had created previously to resolve #'s 1 and 13. The changes on this branch specifically mostly include styling, but some other things were done. Notably, navigating to the root of the app will redirect you to  user creation. Once you create a user, it'll redirect you back to the root where you can make a new shopping list without having to hack any code or anything.

### Setup instructions
This branch will require all devs to create a file in the root of
this project called ".env" and edit it to include an entry for
CLIENT_MAPS_API_KEY=<<< KEY for grocery run CLIENT MAP KEY from Lastpass >>>

We can talk about whether that's the right way to go about it later but that's how I chose to go about it, and given that it's a hackathon and working, probably best to just leave it.

### CODE REVIEW

File: client/components/ListCreationForm.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        const newItems = this.state.items.slice();
+        newItems[metadata.tableData.id] = {
+          ...row,
+          replaceable: String(row.replaceable).toLowerCase() === 'true'
+        };
+
+        this.setState({
+            items: newItems
+        })
+    }
```
[the remark this answers is not in the document]
Comment (JimmyPruitt) [reply client/components/ListCreationForm.js:45]: https://material-table.com/#/docs/features/editable
> To make table editable, you should set editable prop that has onRowAdd, onRowUpdate, onRowDelete functions to manipulate data. Every function must return a promise

[the remark this answers is not in the document]
Comment (addelong) [reply client/components/ListCreationForm.js:45]: Okay that is fine - I'm not sure if setState() returns a promise or not, but even if it does, I still don't think you need the `async` keyword. From purely a language perspective, `async` has no effect if there is not a corresponding `await` keyword inside the function body.

Like, if setState() returns a promise, the callers of these functions could `await` them, and those callers would have to be declared as `async`. But these functions themselves don't need to be declared that way.

[the remark this answers is not in the document]
Comment (JimmyPruitt) [reply client/components/ListCreationForm.js:45]: It does if the consumer of the function explicitly used `.then`, which the maintainer of the Material Table did. They literally have code that goes
`table.onRowAdd(...).then()`, which is why the function must return a promise.

[the remark this answers is not in the document]
Comment (addelong) [reply client/components/ListCreationForm.js:45]: oooooooooooohhhhhhhhh I see where the confusion was here. I didn't know [this](https://stackoverflow.com/a/61120102/276386) about async:

`What may be worth adding that async unconditionally makes your result a Promise.`

TIL

**Comment by addelong** [comment]
merge

**Pull request merged**